src/modules/aws.rs

```diff
@@ -1,15 +1,72 @@
 use std::env;
+use std::fs::File;
+use std::io::{BufRead, BufReader};
+use std::path::PathBuf;
+use std::str::FromStr;
+
+use dirs::home_dir;
 
 use super::{Context, Module};
 
-use crate::config::RootModuleConfig;
+use crate::config::{RootModuleConfig, SegmentConfig};
 use crate::configs::aws::AwsConfig;
 
+fn get_aws_region_from_config(aws_profile: &Option<String>) -> Option<String> {
+    let config_location = env::var("AWS_CONFIG_FILE")
+        .ok()
+        .and_then(|path| PathBuf::from_str(&path).ok())
+        .or_else(|| {
+            let mut home = home_dir()?;
+            home.push(".aws/config");
+            Some(home)
+        })?;
+
+    let file = File::open(&config_location).ok()?;
+    let reader = BufReader::new(file);
+    let lines = reader.lines().filter_map(Result::ok);
+
+    let region_line = if let Some(ref aws_profile) = aws_profile {
+        lines
+            .skip_while(|line| line != &format!("[profile {}]", aws_profile))
+            .skip(1)
+            .take_while(|line| !line.starts_with('['))
+            .find(|line| line.starts_with("region"))
+    } else {
+        lines
+            .skip_while(|line| line != "[default]")
+            .skip(1)
+            .take_while(|line| !line.starts_with('['))
+            .find(|line| line.starts_with("region"))
+    }?;
+
+    let region = region_line.split('=').nth(1)?;
+    let region = region.trim();
+
+    Some(region.to_string())
+}
+
+fn get_aws_region() -> Option<String> {
+    env::var("AWS_DEFAULT_REGION")
+        .or_else(|_| env::var("AWS_REGION"))
```

Comment:
Is the precedent here correct? Should `AWS_REGION` not be prefered if both `AWS_REGION` and `AWS_DEFAULT_REGION` are both set?

Reply:
My mistake! Actually, referring to https://github.com/aws/aws-cli/issues/735, `aws-cli` does not use `AWS_REGION` at all.

Reply:
But I will keep `AWS_REGION` at least precedent because I heard some of the AWS-related programs are still using `AWS_REGION` env var.